src/controllers/QueryController.ts

```diff
@@ -501,11 +503,11 @@ export class QueryController extends RootComponent {
   }
 
   public enableHistory() {
-    this.historyStore = new history.HistoryStore();
+    this.searchInterface.actionHistory = new history.HistoryStore();
   }
 
   public disableHistory() {
-    this.historyStore = new history.HistoryStore(new NullStorage());
+    this.searchInterface.actionHistory = new history.HistoryStore(new NullStorage());
```

Comment:
I moved the actionsHistory instance up to the SearchInterface so that it can be reused inside the Recommendation component.

However, I keep the `enableHistory`, `disableHistory` on the QueryController to avoid the breaking change.
One improvement might be to expose these methods on the SearchInterface itself, and then call them here. e.g.:

```
public enableHistory() {
    this.searchInterface.enableHistory();
}
```

This is more encapsulated, placing the enable/disable methods next to the actionsHistory instance. On the other hand, it also exposes new methods on the SearchInterface, widening the api surface.

Reply:
Couldn't it all be in the QueryController, it is bound to the SearchInterface and already has usageAnalytics as its 3rd constructor argument?

Reply:
Ah, indeed, I placed the `actionHistory` on the `SearchInterface` to make it more direct to access for the `Recommendation` component. However, we could definitely reach it through the `queryController` instance on the `SearchInterface`. Good suggestion! I will make the change,